### ISSUE TITLE
Improve JPEG 2000 format support performance

### ISSUE THREAD

**Issue by tst-ccamp** [google/earthenterprise#423]
When compared to GeoTiff, building an imagery resource based on a JPEG 2000 source image takes many times longer. Investigate possible bottlenecks in the way the library is used. If there isn't something that can be optimized, consider converting the JPEG 2000 file temporarily to GeoTiff before running the rest of the build steps.

**Comment by tst-awankhede** [comment]
with numCPU =  4,

I got the blue marble jp2  =  15 min 
Blue marble tiff = 3 min

Will update the thread once I post it to GDAL group.

**Comment by tst-nfarah** [comment]
A new version of Open Jpeg library was released (http://www.openjpeg.org/2017/08/10/OpenJPEG-2.2.0-released). I tested it in comparison with Open Jpeg 2.1.2 and did see performance improvements that requires setting an env variable to enable multi-threading when decoding the data. The performance improvement (in OpenJpeg 2.2.0) seems to targets mostly tiled jp2 files. Using the new version on 2 CPU, 20 GB RAM and 8 enabled threads, i got:

bluemarble_4km.jp2  (10800 x 5400 pixels). 37.3 mb file size. Single tiled: **7.6 mn** (OpenJpeg 2.2.0) vs **13.8 mn** (OpenJpeg 2.1.2)
usgsLanSat.jp2 (8627 x 7745 pixels). 48 mb file size. Single tiled: **9.9 mn** vs **17.4 mn**

There is ongoing work to speed up more OpenJpeg performance for singled tiled images.
Will be looking forward to re-test the newer version of OpenJpeg once it will be released.
